Formula/watchman.rb

```diff
@@ -23,7 +29,8 @@ def install
                           "--with-pcre",
                           # we'll do the homebrew specific python
                           # installation below
-                          "--without-python"
+                          "--without-python",
+                          "--enable-statedir=#{var}/run/watchman"
```

Comment:
It it's created during build that's pretty much fine. The problems tend to start when it isn't and then the executable is called with root, either via `sudo` or `launchctl` and that directory becomes unwritable for Homebrew.